src/format_rules.ml

```diff
@@ -73,10 +73,18 @@ let gen_rules_output sctx (config : Dune_file.Auto_format.t) ~output_dir =
         let exe = resolve_program "refmt" in
         let args = [Arg_spec.Dep input] in
         Some (Build.run ~dir ~stdout_to:output exe args)
-      | "dune", _ when Dune_file.Auto_format.includes config Dune ->
-        let exe = resolve_program "dune" in
-        let args = [Arg_spec.A "format-dune-file"; Dep input] in
-        Some (Build.run ~dir ~stdout_to:output exe args)
+      | "dune", _ ->
+        begin
+          match Format_dune_lang.parse_file (Some file) with
```

Comment:
Parsing the whole file again feels a bit much. We just need to use `Dune_lexer.is_script` which only inspects the first line.